### PR TITLE
Issue #6

### LINKED ISSUE #6: Queries (RPC and CLI)

### PR DESCRIPTION
**Pull request by nelcolon**
Closes #6

### CODE REVIEW

File: tests/Queries/CLI/account.feature [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    And the arguments of the method has the account address as 4920ce1d787c60e2eaeff366c79e8aa2b82525f1 and height 0
+    Then the CLI response should be detailing:
+        Account:
+            Address:       4920ce1d787c60e2eaeff366c79e8aa2b82525f1
+            Pubkey:        string
+            Coins:         string
+
+  Scenario: Invalid height 
+    Given a pocket core node
+    And a genesis file query-account-rpc-genesis-step12json
```
Comment (luyzdeleon) [inline tests/Queries/CLI/account.feature:17]: Is this the right genesis name?

File: tests/Queries/CLI/account.feature [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    When I use pocket query namespace
+    And I use the method account 
+    And the arguments of the method has the account address as 4920ce1d787c60e2eaeff366c79e8aa2b8252123 and height 0
+    Then the CLI response should be detailing:
+        Account:
+            Address:       null
+            Pubkey:        null
+            Coins:         null
+
+  Scenario: Account number is invalid
```
Comment (luyzdeleon) [inline tests/Queries/CLI/account.feature:42]: This scenario should be called `Account address is invalid`

File: tests/Queries/CLI/node.feature [hunk 3]
```diff
@@ -0,0 +1,54 @@
+Feature: node method in pocket-core cli's query namespace
+
+    Scenario: Returns the height of the pocket core network
```
Comment (luyzdeleon) [inline tests/Queries/CLI/node.feature:3]: This scenario should be called `Returns a pocket network node at the specified height`

File: tests/Queries/CLI/pocket-params.feature [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        And a pocket core network is running
+        When I use query namespace
+        And I use pocket-params method
+        And I use the parameter height 2
+        Then the CLI response should be detailing: 
+         "claim_expiration":"100",
+            "proof_waiting_period":"3",
+            "replay_attack_burn_multiplier":"0",
+            "session_node_count":"5",
+            "supported_blockchains":[
```
Comment (luyzdeleon) [inline tests/Queries/CLI/pocket-params.feature:14]: Shorten this to 3 blockchains, it's already a list, no need to make it so long.

File: tests/Queries/CLI/pocket-params.feature [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+                "0062",
+                "0063",
+                "0064",
+                "0065",
+                "0066",
+                "0067",
+                "0068",
+                "0069",
+                "006A",
+                "006B"
```
Comment (luyzdeleon) [inline tests/Queries/CLI/pocket-params.feature:121]: Missing the last `]`

File: tests/Queries/CLI/pocket-params.feature [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+    Scenario: Return error with incorrect height
+        Given a pocket core node
+        And a genesis file query-pocketparams-genesis-step1.json
+        And a pocket core network is running
+        When I use query namespace
+        And I use pocket-params method
+        And I use the parameter height 99999999
+        Then the CLI response should be detailing: 
+            "strconv.Atoi: parsing  \"99999999\": value out of range"
```
Comment (luyzdeleon) [inline tests/Queries/CLI/pocket-params.feature:131]: The error message shouldn't be value of range, as `99999999 ` is a possible block height, it should be something like: `Invalid block height, it should be less or equal than the current block height`

File: tests/Queries/CLI/daoOwner.feature [hunk 7]
```diff
@@ -0,0 +1,11 @@
+Feature: daoOwner method in pocket-core cli's query namespace
+    Scenario: Get daoOwner
+        Given a pocket core node
+        And a genesis file query-daoOwner-genesis-step1.json
+        And a pocket core network is running
+        When I use the pocket namespace query
+        And I use the method of daoOwner
+        Then the CLI should show a response of:
+            "daoOwnerAddr"
+
```
Comment (emedrano2112) [inline tests/Queries/CLI/daoOwner.feature:10]: Remove extra spaces

File: tests/Queries/CLI/node-params.feature [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        And a genesis file query-nodeparams-rpc-genesis-step3.json
+        And a node doesn't exist in the pocket network in height -i
+        When I use pocket's namespace query
+        And the parameter I use is height -i 
+        Then the CLI response will be a JSON detail indicating:
+        {
+            "code":400,
+            "message":"invalid character 'i' in numeric literal"
+        }
+
```
Comment (emedrano2112) [inline tests/Queries/CLI/node-params.feature:52]: Remove extra spaces

File: tests/Queries/CLI/node-receipts.feature [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        And a node doesn't exist in the pocket core node with the address 0xA5DE6D4184016708c1040c355F1c958192276DB5
+        When I use namespace query 
+        And I use method node-receipts
+        And I send parameters of address 0xA5DE6D4184016708c1040c355F1c958192276DB5 and height 2
+        Then the CLI response will be a JSON indicating:
+        {
+            "null"
+        }  
+    
+
```
Comment (emedrano2112) [inline tests/Queries/CLI/node-receipts.feature:50]: Remove extra spaces

File: tests/Queries/CLI/node.feature [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+        When I use pocket query namespace
+        And I use the method node
+        And I use the parameters address 02$%0DA72BBCA5DC9792C72AC3A2616910C479B3 and height 2
+        Then the CLI should show a JSON response indicating:
+        {
+            "code":400,
+            "message":"encoding/hex: invalid byte: U+0024 '$'"
+        }    
+    
+
```
Comment (emedrano2112) [inline tests/Queries/CLI/node.feature:53]: Remove extra spaces

File: tests/Queries/CLI/supply.feature [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+        Then the CLI response will be indicating:    
+        {
+            "code":400,
+            "message":"{
+                "codespace":"sdk",
+                "code":1,
+                "message":"failed to load state at height -1; failed to get Store: no data (latest height: 4308)"
+            }"
+        }
+
```
Comment (emedrano2112) [inline tests/Queries/CLI/supply.feature:35]: Remove extra spaces

File: tests/Queries/RPC/apps.feature [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+            }
+        }
+        Then the HTTP response will be a null detail indicating:
+        {
+            "result":null,
+            "total_pages":1,
+            "page":1
+        }
+    
+
```
Comment (emedrano2112) [inline tests/Queries/RPC/apps.feature:57]: Remove extra spaces

File: tests/Queries/RPC/balance.feature [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+        }
+        Then the HTTP response will be a balance detailing:
+        {
+            "balance": 1000000000
+        }
+    
+    Scenario: account doesn't exist
+        Given a pocket core node
+        And a genesis file query-balance-rpc-genesis-step2.json
+        And an account exists in the pocket core network with address 1b1973906ee85993e994422eddeab89f385a1234
```
Comment (emedrano2112) [inline tests/Queries/RPC/balance.feature:20]: An account "doesn't exist" is maybe what you'd like to say? If so, then if an account doesn't exist, the query returns balance: 0?

File: tests/Queries/RPC/nodeparams.feature [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+        {
+            "height": -i
+        }
+        Then the HTTP response will be an account detail indicating:
+        {
+            "code":400,
+            "message":"invalid character 'i' in numeric literal"
+        }
+    
+
```
Comment (emedrano2112) [inline tests/Queries/RPC/nodeparams.feature:61]: Remove extra spaces

File: tests/Queries/RPC/nodereceipt.feature [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+            "code":400,
+            "message":
+            "{
+                "codespace":"sdk",
+                "code":1,
+                "message":"type in the receipt query is not recognized: (relay or challenge)"
+            }"
+        }
+
+
```
Comment (emedrano2112) [inline tests/Queries/RPC/nodereceipt.feature:56]: Remove extra spaces

File: tests/Queries/RPC/nodereceipts.feature [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+            "address": "0xA5DE6D4184016708c1040c355F1c958192276DB5",
+            "height": 2
+        }
+        Then the HTTP response will be a nodes detail indicating:
+        {
+            "null"
+        }  
+    
+    
+
```
Comment (emedrano2112) [inline tests/Queries/RPC/nodereceipts.feature:60]: Remove extra spaces
